packages/cli/src/commands/split/index.ts

```diff
@@ -237,49 +242,73 @@ function gatherComponentsFiles(
   componentsFiles[componentType][componentName] = { inherits, filename };
 }
 
-function iteratePaths(
-  openapi: Oas3Definition,
-  pathsDir: string,
-  openapiDir: string
+function iteratePathItem(
+  openapi: any,
+  openapiDir: string,
+  componentsFiles: object,
+  rootDocObj: string,
+  customDir?: string,
 ) {
-  const { paths } = openapi;
-  if (paths) {
-    for (const oasPath of Object.keys(paths)) {
-      const pathFile = path.join(pathsDir, pathToFilename(oasPath)) + '.yaml';
-      const pathData: Oas3PathItem = paths[oasPath] as Oas3PathItem;
+  const docObj =
+    rootDocObj === PATHS
+      ? openapi.paths
+      : openapi[WEBHOOKS]
+      ? openapi[WEBHOOKS]
+      : openapi[xWEBHOOKS];
+
+  if (docObj) {
+    const checkDir = customDir ? customDir : path.join(openapiDir, rootDocObj);
+    fs.mkdirSync(checkDir, { recursive: true });
+
+    for (const oasPath of Object.keys(docObj)) {
+      const pathFile = path.join(
+        checkDir,
+        `${rootDocObj === PATHS ? pathToFilename(oasPath) : oasPath}.yaml`,
+      );
+      const pathData: Oas3PathItem = docObj[oasPath] as Oas3PathItem;
 
       for (const method of OPENAPI3_METHOD_NAMES) {
         const methodData = pathData[method];
         const methodDataXCode = methodData?.['x-code-samples'] || methodData?.['x-codeSamples'];
-        if (!methodDataXCode || !Array.isArray(methodDataXCode)) { continue; }
+        if (!methodDataXCode || !Array.isArray(methodDataXCode)) {
+          continue;
+        }
         for (const sample of methodDataXCode) {
           if (sample.source && (sample.source as any).$ref) continue;
           const sampleFileName = path.join(
             openapiDir,
             'code_samples',
             sample.lang,
-            pathToFilename(oasPath),
-            method + langToExt(sample.lang)
+            docObj === PATHS ? pathToFilename(oasPath) : oasPath,
+            method + langToExt(sample.lang),
           );
 
           fs.mkdirSync(path.dirname(sampleFileName), { recursive: true });
           fs.writeFileSync(sampleFileName, sample.source);
           // @ts-ignore
           sample.source = {
-            $ref: slash(path.relative(pathsDir, sampleFileName))
+            $ref:
+              rootDocObj === PATHS
```

Comment:
maybe a good point to move it to function

Reply:
Done - https://github.com/Redocly/openapi-cli/blob/split-webhooks/packages/cli/src/commands/split/index.ts#L352